test/examples/uct_hello_world.c

```diff
@@ -1,17 +1,566 @@
+                          (void **)&peer_iface);
+        CHKERR_JUMP(0 != status, "ifaces exchange", out_free_if_addrs);
+    }
+
+    own_ep = (uct_ep_addr_t*)calloc(1, if_info.attr.ep_addr_len);
+    CHKERR_JUMP(NULL == own_ep, "allocate memory for ep addrs", out_free_if_addrs);
+
+    if (if_info.attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP) {
+        /* Create new endpoint */
+        status = uct_ep_create(if_info.iface, &ep);
+        CHKERR_JUMP(UCS_OK != status, "create endpoint", out_free_ep_addrs);
+
+        /* Get endpoint address */
+        status = uct_ep_get_address(ep, own_ep);
+        CHKERR_JUMP(UCS_OK != status, "get endpoint address", out_free_ep);
+    }
+
+    status = sendrecv(oob_sock, own_ep, if_info.attr.ep_addr_len,
+                      (void **)&peer_ep);
+    CHKERR_JUMP(0 != status, "EPs exchange", out_free_ep);
+
+    if (if_info.attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP) {
+        /* Connect endpoint to a remote endpoint */
+        status = uct_ep_connect_to_ep(ep, peer_dev, peer_ep);
+        barrier(oob_sock);
+    } else if (if_info.attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) {
+        /* Create an endpoint which is connected to a remote interface */
+        status = uct_ep_create_connected(if_info.iface, peer_dev, peer_iface, &ep);
+    } else {
+        status = UCS_ERR_UNSUPPORTED;
+    }
+    CHKERR_JUMP(UCS_OK != status, "connect endpoint", out_free_ep);
+
+    /*Set active message handler */
+    status = uct_iface_set_am_handler(if_info.iface, id, hello_world,
+                                      &cmd_args.func_am_type,
+                                      UCT_AM_CB_FLAG_SYNC);
+    CHKERR_JUMP(UCS_OK != status, "set callback", out_free_ep);
+
+    if (cmd_args.server_name) {
+        char *str = (char *)malloc(cmd_args.test_strlen);
+        generate_random_string(str, cmd_args.test_strlen);
+
+        if (cmd_args.func_am_type == FUNC_AM_SHORT) {
+            am_short_args_t send_args;
+            am_short_params_pack(str, cmd_args.test_strlen, &send_args);
+            /* Send active message to remote endpoint */
+            status = uct_ep_am_short(ep, id, send_args.header,
+                                     send_args.payload, send_args.len);
+        } else if (cmd_args.func_am_type == FUNC_AM_BCOPY) {
+            am_bcopy_args_t args = {
+                .data = str,
+                .len = cmd_args.test_strlen
+            };
+            /* Send active message to remote endpoint */
+            status = uct_ep_am_bcopy(ep, id, am_bcopy_data_pack_cb, &args);
+            if (status >= 0) {
+                status = UCS_OK;
+            }
+        } else if (cmd_args.func_am_type == FUNC_AM_ZCOPY) {
+            uct_mem_h memh;
+            status = uct_md_mem_reg(if_info.pd, str, cmd_args.test_strlen, 0, &memh);
+            CHKERR_JUMP(UCS_OK != status, "memory registration", out_free_ep);
+            const uct_iov_t iov = {
```

Comment:
I'm going to move short/bcopy/zcopy to separate fucntions, So, this and other blocks like this will be placed in the beginning of corresponding function